justifi-tech/web-component-library
# FE-258 iframe resize issue

PR related ([payments-js](https://github.com/justifi-tech/payments-js/pull/108))

Resolved an issue where the iframe did not resize correctly upon initial loading within an element set to display: none ([FE-258](https://justifi-ai.atlassian.net/jira/software/c/projects/FE/boards/16?selectedIssue=FE-258)).

This was fixed by integrating the iframe-resizer library.

It automatically handles the resize of the iframe dispensing the use of the resize method.

Adds deprecation warning messages for the resize method




Links
-----



Developer considerations
--------------

  - If this is a MAJOR version, what are the breaking API changes? Was anything added to the API?
  - If this is a MINOR version, what feature was added? Why would consumers want this new version?
  - If this is a PATCH version, what was fixed?


Steps for Testing/QA
--------------------

* This fix depends on changes in both `payments-js` and `web-components-library`
* This can be tested using the HTML provided on the attachments of [this issue](https://justifi-ai.atlassian.net/jira/software/c/projects/FE/boards/16?selectedIssue=FE-258)
* Pull the `main` branch of `web-components-library`, run `npm install`, build it and run it locally.
* Pull also the `payments-js` `main` branch, run `yarn install`  and run it.
* In the HTML mentioned above, change the  `<script>` tag to point to the web-components running locally.
```
<script
      type="module"
      src="/web-component-library/stencil-library/dist/webcomponents/webcomponents.esm.js"
    ></script>
```
* In order to open correctly the HTML on the browser you should start a web-server on the folder the HTML file was created.
* Install `npm install -g http-server` and then run `http-server` in the same folder of the HTML file.
* Then you can access something like `http://127.0.0.1:8080/test.html` and check.
* The card-form should appear after 3 seconds (timeout) correctly without being cut off by the container.






[FE-258]: https://justifi-ai.atlassian.net/browse/FE-258?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
